### PR TITLE
[Slim4] Data mocker for scalar types

### PR DESCRIPTION
**Pull request by ybelenko**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

I'm working on mock server for Slim generator. Maybe my `OpenApiDataMockerInterface` can be useful in other PHP generators. Right now only scalar data types implemented. ~~`array` and `object` are coming next. Then I'll add `enum` option to `mockString` and method `mockSchema` to generate data from spec schema object.~~(will be implemented in next PRs)

Related to #3545

cc @jebentier, @dkarlovi, @mandrean, @jfastnacht, @ackintosh, @renepardon

### CODE REVIEW

**Comment by ybelenko** [comment]
> Right now only scalar data types implemented. `array` and `object` are coming next. Then I'll add `enum` option to `mockString` and method `mockSchema` to generate data from spec schema object.

I've reconsidered. Small PR will be easier to review. Let's keep this PR with scalar types only, I think it's a good starting point. I'll work on other types in local branch and submit when this will be merged.

**Comment by ybelenko** [comment]
### How to test

1. Checkout into feature branch
```shell
$ git checkout -b ybelenko-slim_data_mocker master
$ git pull https://github.com/ybelenko/openapi-generator.git slim_data_mocker
```
2. Generate new Slim4 samples and go to just created build
```shell
$ mvn clean package
$ bin/php-slim4-server-petstore.sh
$ cd samples/server/petstore/php-slim4
```
3. Start Composer installation and run unit tests
```shell
$ composer install
$ composer test
$ composer phpcs
```

File: samples/server/petstore/php-slim4/lib/Mock/OpenApiDataMockerInterface.php [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+     *
+     * @throws \InvalidArgumentException when invalid arguments passed
+     *
+     * @return string
+     */
+    public function mockString(
+        $dataFormat = null,
+        $minLength = 0,
+        $maxLength = null
+    );
```
Comment (ybelenko) [inline samples/server/petstore/php-slim4/lib/Mock/OpenApiDataMockerInterface.php]: Should I add `$pattern = null` argument to `mockString()` method? I won't implement it right away, but it helps to overcome breaking changes in future.

Comment (wing328) [reply samples/server/petstore/php-slim4/lib/Mock/OpenApiDataMockerInterface.php]: Sounds like a good idea to me and we can leave a `TODO` next to it.

Besides pattern, here are some other useful properties:
```
maximum
exclusiveMaximum
minimum
exclusiveMinimum
maxLength
minLength
pattern (This string SHOULD be a valid regular expression, according to the ECMA 262 regular expression dialect)
maxItems
minItems
uniqueItems
maxProperties
minProperties
```



Ref: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#properties

**Pull request merged**